src/components/builder/values/referentielijsten/utils.ts

```diff
@@ -0,0 +1,25 @@
+import {IntlShape} from 'react-intl';
+
+import {
+  ReferentielijstenTabelItem,
+  ReferentielijstenTabelOption,
+} from '@/components/builder/values/referentielijsten/types';
+
+// Transform options to the desired shape for the options of the Select component
+export function transformItems(
+  items: ReferentielijstenTabelItem[] | ReferentielijstenTabelOption[],
+  intl: IntlShape
+) {
+  return items.map(item => {
+    const {code, naam, isGeldig} = item;
+    return {
+      value: code,
+      label: !isGeldig
+        ? `${naam} ${intl.formatMessage({
+            description: 'Message to indicate that Referentielijsten tabel is expired',
+            defaultMessage: '(niet meer geldig)',
+          })}`
+        : naam,
```

Comment:
The correct way would be to incorporate the label directly in the message so that order of sentences can be adapted if needed:

```suggestion
      label: intl.formatMessage({
        description: 'Label for referentielijsten table or item',
        defaultMessage: '{naam}{isValid, select, true {} other { (no longer valid)}}',
      }, {name: naam, isValid: isGeldig})
```
Reference for message syntax: https://formatjs.github.io/docs/core-concepts/icu-syntax/